user_tools/src/spark_rapids_pytools/cloud_api/dataproc.py

```diff
@@ -158,9 +158,18 @@ def get_matching_executor_instance(self, cores_per_executor):
         executors_from_config = self.configs.get_value('clusterInference', 'defaultCpuInstances', 'executor')
         # TODO: Currently only single series is supported. Change this to a loop when using multiple series.
         series_name, unit_info = list(executors_from_config.items())[0]
+        # If the number of cores is a valid entry in the list, return the equivalent instance
         if cores_per_executor in unit_info['vCPUs']:
             return f'{series_name}-{cores_per_executor}'
-        return None
+        # If the num-cores is not in the list, then we need to adjust the cores to the best match.
+        # The loop below should always return a valid instance type. It takes the instance type with
+        # number of cores less than or equal to the "cores_per_executor".
```

Comment:
why do we chose less here?  comment should explain.

I'm not sure that is the best solution actually.  If say I somehow had 36 cores and the node type I chose had 8, that is quite different.  But if the next bigger node is 48 from performance point of view that would likely be a better match.  But its going to be more expensive.  

We may want to ask for some input on this but I would rather see us see how far off the smallest/largest is and go with the closest one. Then log a clear message on that is what we did.